### DISCUSSION TITLE
[BP#14] Créer une architecture applicative modulaire

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#74] [Bonnes pratiques] [answered]
Cette BP me semble intéressante mais pas assez objective/trop générique en l'état.
Peut-être serait-il intéressant d'expliciter les BP d'architecture modulaire sous-entendues ? Peut-être en découpant en plusieurs BP?

**Comment by didierfred** [comment]
Attention aussi au cas des micro-services , par exemple  découper une application java en 10 micro services  conduit à : 
 - augmenter fortement l'empreinte mémoire (chaque microservice montant en mémoire les mêmes libs) 
 - ajouter des communications réseau entre chaque services 
 
Sans préjuger de l'intéret des micro-services , j'ai trop tendance à voir des modularités inutiles

**Reply by devcrafting** [reply]
Oui j'avais aussi cela en tête, probablement des BP à ajouter autour des microservices 😉

**Comment by ACTLEM** [comment]
Assez d'accord pour améliorer cette BP. Et 100% ok avec @didierfred, il faut trouver le bon compromis. Je vois plus cette BP comme un rappel qu'il faut mettre en place une architecture scalable fonctionnellement (éviter ainsi de toute refaire quand il y a un changement fonctionnelle) et techniquement (adapté aux contraintes techniques courantes et facilement réajustables).

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: Ok pour modification avec clarification de la modularité (et peut-être un avertissement sur les micro services), et la notion de scalabilité

Issue: https://github.com/cnumr/best-practices/issues/178

**Comment by devcrafting** [comment]
Faire référence au choix des technologies #199 pouvant être différencié en fonction du contexte => analyse fonctionnelle préalable du domaine métier et découpage en contextes (notions DDD)
